src/sanity_checker.rs

```diff
@@ -0,0 +1,220 @@
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.listen_tcp_acars,
+        "AR_LISTEN_TCP_ACARS/--listen-tcp-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.receive_tcp_acars,
+        "AR_RECEIVE_TCP_ACARS/--receive-tcp-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.listen_udp_vdlm2,
+        "AR_LISTEN_UDP_VDLM2/--listen-udp-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.listen_tcp_vdlm2,
+        "AR_LISTEN_TCP_VDLM2/--listen-tcp-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.receive_tcp_vdlm2,
+        "AR_RECEIVE_TCP_VDLM2/--receive-tcp-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.send_udp_acars,
+        "AR_SEND_UDP_ACARS/--send-udp-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.send_udp_vdlm2,
+        "AR_SEND_UDP_VDLM2/--send-udp-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.send_tcp_acars,
+        "AR_SEND_TCP_ACARS/--send-tcp-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.send_tcp_vdlm2,
+        "AR_SEND_TCP_VDLM2/--send-tcp-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.receive_zmq_vdlm2,
+        "AR_RECEIVE_ZMQ_VDLM2/--receive-zmq-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid_with_host(
+        &config_options.receive_zmq_acars,
+        "AR_RECEIVE_ZMQ_ACARS/--receive-zmq-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.serve_tcp_acars,
+        "AR_SERVE_TCP_ACARS/--serve-tcp-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.serve_tcp_vdlm2,
+        "AR_SERVE_TCP_VDLM2/--serve-tcp-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.serve_zmq_acars,
+        "AR_SERVE_ZMQ_ACARS/--serve-zmq-acars",
+    ) {
+        is_input_sane = false;
+    }
+
+    if !check_ports_are_valid(
+        &config_options.serve_zmq_vdlm2,
+        "AR_SERVE_ZMQ_VDLM2/--serve-zmq-vdlm2",
+    ) {
+        is_input_sane = false;
+    }
+
+    match is_input_sane {
+        true => Ok(()),
+        false => Err("Config option sanity check failed".to_string()),
+    }
+}
+
+fn check_ports_are_valid(option_ports: &Option<Vec<String>>, name: &str) -> bool {
```

Comment:
Now that I'm a little more familiar with the code base, I'm wondering if we want to change any of the option switches that _only_ accept ports from `Option<Vec<String>>` to `Option<Vec<u16>>` instead?

This would mean that your max port number cannot be bigger than 65535, as it's the MAX size for a u16, so you only have to check that it's not zero. It'll cut down a bunch of the code needed to validate your config here.

Reply:
That makes a ton of sense